modules/local/samtools_bowtie.nf

```diff
@@ -0,0 +1,35 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process SAMTOOLS_BOWTIE {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::samtools=1.10" : null)
+    if (workflow.containerEngine == 'singularity' && !params.pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/samtools:1.10--h9402c20_2"
+    } else {
+        container "quay.io/biocontainers/samtools:1.10--h9402c20_2"
+    }
+
+    input:
+    tuple val(meta), path(sam)
+    
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path  "*.version.txt"         , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    samtools view -bS -o ${prefix}.bam $sam
+    echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' > samtools.version.txt
+    """
+}
```

Comment:
Why is the file called `samtools_bowtie` and not just `samtools`, if it only checks the samtools version?

Reply:
I think because it was a samtools module specifically used for the bowtie version of the pipeline, but I agree the name is bad. Probably something like `samtools_samtobam` is better here :+1: